### PR TITLE
Hide Locale column and grouping option when i18n plugin is not installed

### PR DESCRIPTION
**Pull request by simotae14**
### What does it do?

It checks if the i18n plugin is installed, if not we need to hide the Locale column in the Release Details page and also the Grouping locale option.

### Why is it needed?

We want to replicate a similar behavior as in the CM. In the Release Details we can have entries from multiple collections so it is not possible to hide the column when a Collection is not using the i18n plugin (and also the entries can be grouped by Locale or actions).

### How to test it?

Check a Release Details page with the i18n plugin installed and then remove it with
```
yarn remove @strapi/plugin-i18n 
```
and re run the application, the locale option in the Grouping dropdown and the locale column need to be removed

### Related issue(s)/PR(s)

CS-549

### CODE REVIEW

**Review by joshuaellis — CHANGES_REQUESTED** [review]
I'm not sure about this @simotae14 this goes against how plugins work interacting with core plugins. IMO we should be injecting this from the i18n plugin.

What was the reasoning for doing it this way round?

File: packages/core/admin/admin/src/StrapiApp.tsx
Comment (joshuaellis) [inline packages/core/admin/admin/src/StrapiApp.tsx:1]: Would it make more sense to add `createHook` to the bootstrap function of plugins, have the releases plugin "create" it's hook so again, it's locked to the plugin?

**Review by joshuaellis — COMMENTED** [review]
I think this is highlighting the need for a generic table API that can apply across the whole application 🤔 not suggesting you do this now of course, but long term – what do you think?

**Comment by simotae14** [comment]
Yes, I agree, there are different places where we can inject or remove columns based on conditions

**Review by joshuaellis — APPROVED** [review]
Not tested, but code changes 🌟

**Comment by remidej** [comment]
I'm not able to test it, after uninstalling i18n I get this error:

```sh
TypeError: Cannot read properties of undefined (reading 'service')
at disableContentTypeLocalized (strapi/packages/core/content-releases/server/src/migrations/index.ts:197:51)
```

**Review by remidej — APPROVED** [review]
the problem is in the backend, but I can confirm the frontend part works as expected

**Pull request merged**